assets/builtin-policy.csv: no such file or directory when trying to use `argo-cd/util/clusterauth` as dependency

Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**
I am using `github.com/argoproj/argo-cd/util/clusterauth` in my code as a dependency. When I try to run the docker image for this code in Kubernetes, I get the following error:
```
panic: stat /go/pkg/mod/github.com/argoproj/argo-cd@v1.3.6/assets/builtin-policy.csv: no such file or directory

goroutine 1 [running]:
github.com/argoproj/argo-cd/util/assets.init.0()
	/go/pkg/mod/github.com/argoproj/argo-cd@v1.3.6/util/assets/assets.go:19 +0x2de
```

Here's my docker file
```
# Build the manager binary
FROM golang:1.13 as builder

WORKDIR /workspace
# Copy the Go Modules manifests
COPY go.mod go.mod
COPY go.sum go.sum
# cache deps before building and copying source so that we don't need to re-download as much
# and so that source changes don't invalidate our downloaded layer
RUN go mod download

# Copy the go source
COPY main.go main.go
# COPY api/ api/
COPY controllers/ controllers/

# Build
RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go

# Use distroless as minimal base image to package the manager binary
# Refer to https://github.com/GoogleContainerTools/distroless for more details
FROM gcr.io/distroless/static:nonroot
WORKDIR /
COPY --from=builder /workspace/manager .
USER nonroot:nonroot

ENTRYPOINT ["/manager"]

```
This works locally without any problem (because `assets` are present in my local `$GOPATH`). It also works if I run it inside the `builder` container itself (`assets` are available here as well) but it does not work if I try to run it as a standalone binary in the final container. This is because `/go/pkg/mod/github.com/argoproj/argo-cd@v1.3.6/assets/builtin-policy.csv` is not present in the final container. `clusterauth` dependency is dependent on `util/rbac` which in turn is dependent on `util/assets`, which throws the above error because no `builtin-policy.csv` is present. 
https://github.com/argoproj/argo-cd/blob/dbcfccbfc0c6d17ecbd4a2fd2508a78d651cf44f/util/assets/assets.go#L17

**To Reproduce**

1. Include `argo-cd/util/clusterauth` as dependency in your code and try to build a docker image with it as described in the dockerfile above (use an intermediate image to build the binary and copy it into the a separate container to run it as an image)

**Expected behavior**

The dependency should be resolved properly. 

**Version**

```shell
argocd: v1.3.6+89be1c9
  BuildDate: 2019-12-10T22:46:45Z
  GitCommit: 89be1c9ce6db0f727c81277c1cfdfb1e385bf248
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.3.6+89be1c9
  BuildDate: 2019-12-10T22:47:48Z
  GitCommit: 89be1c9ce6db0f727c81277c1cfdfb1e385bf248
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: Version: {Version:kustomize/v3.2.1 GitCommit:d89b448c745937f0cf1936162f26a5aac688f840 BuildDate:2019-09-27T00:10:52Z GoOs:linux GoArch:amd64}
  Helm Version: v2.15.2
  Kubectl Version: v1.14.0
```

**Pod Logs**

```
panic: stat /go/pkg/mod/github.com/argoproj/argo-cd@v1.3.6/assets/builtin-policy.csv: no such file or directory

goroutine 1 [running]:
github.com/argoproj/argo-cd/util/assets.init.0()
	/go/pkg/mod/github.com/argoproj/argo-cd@v1.3.6/util/assets/assets.go:19 +0x2de
```

Related: #2592 This issue seems to fix the problem with `api` client but it persists in `clusterauth`.

--- a/assets/tools.go
+++ b/assets/tools.go
@@ -1,0 +1,4 @@
+// +build tools
+
+// This existence of this package allows vendoring of the assets in this directory by go 1.13+.
+package tools

